tests/test_valve.py

```diff
@@ -0,0 +1,163 @@
+        mock.return_value = RESPONSE200
+
+        valveid = uuid.uuid4()
+
+        self.valve.get_valve(valveid)
+
+        args, kwargs = mock.call_args
+
+        # Check that the mock function is called with the rights args.
+        self.assertEqual(args[1], f"{VALVE_API_URL}/valve/getValve/{valveid}")
+        self.assertEqual(args[0], "GET")
+        self.assertEqual(kwargs["data"], None)
+
+    @patch("requests.Session.request")
+    def test_get_base_station(self, mock):
+        """Test if the get_base_station method works as expected."""
+        mock.return_value = RESPONSE200
+
+        baseid = str(uuid.uuid4())
+
+        self.valve.get_base_station(baseid)
+
+        args, kwargs = mock.call_args
+
+        # Check that the mock function is called with the rights args.
+        self.assertEqual(
+            args[1], f"{VALVE_API_URL}/valve/getBaseStation/{baseid}"
+        )
+        self.assertEqual(args[0], "GET")
+        self.assertEqual(kwargs["data"], None)
+
+    @patch("requests.Session.request")
+    def test_list_base_stations(self, mock):
+        """Test if the list_base_stations method works as expected."""
+        mock.return_value = RESPONSE200
+
+        userid = str(uuid.uuid4())
+
+        self.valve.list_base_stations(userid)
+
+        args, kwargs = mock.call_args
+
+        # Check that the mock function is called with the rights args.
+        self.assertEqual(
+            args[1], f"{VALVE_API_URL}/valve/listBaseStations/{userid}"
+        )
+        self.assertEqual(args[0], "GET")
+        self.assertEqual(kwargs["data"], None)
+
+    @patch("requests.Session.request")
+    def test_list_valves(self, mock):
+        """Test if the list_valves method works as expected."""
+        mock.return_value = RESPONSE200
+
+        baseid = str(uuid.uuid4())
+
+        self.valve.list_valves(baseid)
+
+        args, kwargs = mock.call_args
+
+        # Check that the mock function is called with the rights args.
+        self.assertEqual(args[1], f"{VALVE_API_URL}/valve/listValves/{baseid}")
+        self.assertEqual(args[0], "GET")
+        self.assertEqual(kwargs["data"], None)
+
+    @patch("requests.Session.request")
+    def test_set_default_runtime(self, mock):
+        """Test if the set_default_runtime method works as expected."""
+        mock.return_value = RESPONSE200
+
+        valveid = str(uuid.uuid4())
+        duration = randrange(86400)
+
+        self.valve.set_default_runtime(valveid, duration)
+
+        args, kwargs = mock.call_args
+
+        # Check that the mock function is called with the rights args.
+        self.assertEqual(args[1], f"{VALVE_API_URL}/valve/setDefaultRuntime")
+        self.assertEqual(args[0], "PUT")
+        self.assertEqual(
+            kwargs["data"],
+            json.dumps(
+                {"valveId": valveid, "defaultRuntimeSeconds": duration}
+            ),
+        )
+
+        # Check that values should be within range.
```

Comment:
I would make these separate test methods to explicitly have a 'happy flow' test and 'exception tests'. Could you process that?

Reply:
Let me know if that's what you meant, happy to fix if not.